### ISSUE TITLE
The version of the resource is not being updated when using a fixed environment name.

### ISSUE THREAD

**Issue by joshzarrabi** [ljfranklin/terraform-resource#67]
We want to use the terraform resource to spin up environments with a constant name, and whenever we do a put, we seem to be getting the same version of the resource. It is 
```
env_name: <env-name>
serial: 2
```

So, when we try to use the `get` of the resource on a worker that we did not `put` on, we are getting whichever version of the resource is cached on the worker, as concourse thinks that it has the latest version already. The version is correct on the worker that did the `put` because concourse forces a `get` on that worker every time. We had to change to use random environment names to get around the issue. 

We are using the `beta-backend` feature with gcp as our backend.

Thanks!

**Comment by ljfranklin** [comment]
Is the terraform apply a no-op, e.g. `Apply complete! Resources: 0 added, 0 changed, 0 destroyed.`? In that case the serial probably doesn't change.

**Comment by joshzarrabi** [comment]
No, it is creating a new environment,

**Comment by ljfranklin** [comment]
Oh, so you're creating a new environment with the same name each time but you're destroying it before the next pipeline run. Don't think I considered that use case, I'll have to think about how to fix this in the resource. In the meantime I think the main workarounds are having the resource generate a random name or not running terraform destroy at the end of the run.

**Comment by joshzarrabi** [comment]
Yeah that's what we are trying to do.  Okay, we can just use random names. Thanks!

**Comment by ljfranklin** [comment]
Looks like I need to have the resource check the `lineage` field in the state file as well as the serial: https://www.terraform.io/docs/backends/state.html#manual-state-pull-push. I'll try to get this fixed in a few days. Thanks for the heads up!

**Comment by joshzarrabi** [comment]
Thank you!
